Description: Use system libsodium
 This patch should be enhanced to use a system libsodium, if present, otherwise
 fallback to the included copy.  Then it can be submitted upstream.
Forwarded: no
Origin: vendor
Author: Richard Laager <rlaager@wiktel.com>
Last-Update: 2016-11-29
--- a/libntp/wscript
+++ b/libntp/wscript
@@ -49,14 +49,14 @@
 	includes = [
 		"%s/libisc/include/" % srcnode,
 		"%s/libisc/unix/include/" % srcnode, # XXX: platform: requires unix/win32 switch.
-		"%s/libsodium/include/" % srcnode,
+		"/usr/include/sodium",
 	]
 
 	# C library
 	ctx(
 		target		= "ntp",
 		features	= "c cstlib bld_include src_include",
-		use		= "sodium",
+		use		= "SODIUM",
 		source		= libntp_source + libntp_source_sharable,
 		includes	= includes,
 	)
--- a/ntpd/wscript
+++ b/ntpd/wscript
@@ -129,11 +129,11 @@
 		target		= "ntpd",
 		features	= "c rtems_trace cprogram bld_include src_include libisc_include libisc_pthread_include ntp_version",
 		source		= ntpd_source,
-		use		= "libntpd_obj isc ntp sodium M parse RT CAP SECCOMP PTHREAD CRYPTO DNS_SD DNS_SD_INCLUDES %s" % use_refclock,
+		use		= "libntpd_obj isc ntp SODIUM M parse RT CAP SECCOMP PTHREAD CRYPTO DNS_SD DNS_SD_INCLUDES %s" % use_refclock,
 		includes	= [
 					"%s/host/ntpd/" % ctx.bldnode.parent.abspath(),
 					"%s/ntpd/" % srcnode,
-                        "%s/libsodium/include" % srcnode
+					"/usr/include/sodium"
 		],
 		install_path	= ntpd_install_path,
 	)
--- a/wafhelpers/configure.py
+++ b/wafhelpers/configure.py
@@ -287,6 +287,7 @@
 
         ctx.check_cc(lib="m", comment="Math library")
         ctx.check_cc(lib="rt", mandatory=False, comment="realtime library")
+        ctx.check_cc(lib="sodium", mandatory=True, comment="sodium library")
 
         # Find OpenSSL. Must happen before function checks
         if ctx.options.enable_crypto:
